crates/solver/src/solver/single_order_solver.rs

```diff
@@ -202,6 +202,115 @@ impl SingleOrderSolver {
         }
     }
 
+    async fn solve_instrumented(
```

Comment:
Naming nit: this future is not instrumented, it is just an inner function that can be instrumented. I would change the name. That being said, if you apply the suggestion from https://github.com/cowprotocol/services/pull/1522/files#r1199135279, then you wouldn't need this method anymore anyway.